src/components/molecules/Visualizer/Engine/Cesium/hooks.ts

```diff
@@ -163,6 +191,206 @@ export default ({
+      Cartographic.fromCartesian(north),
+      ellipsoid,
+    );
+    const geodesicHorizontal = new EllipsoidGeodesic(
+      CartographicCenterPoint,
+      Cartographic.fromCartesian(east),
+      ellipsoid,
+    );
+    return { geodesicVertical, geodesicHorizontal };
+  }, [property?.cameraLimiter?.cameraLimitterTargetArea]);
+
+  // calculate inner limiter dimensions
+  const targetWidth = 1000000;
+  const targetHeight = 1000000;
+  const limiterDimensions = useMemo(():
+    | undefined
+    | {
+      cartographicDimensions: {
+        rightDemention: Cartographic;
+        leftDemention: Cartographic;
+        topDemention: Cartographic;
+        bottomDemention: Cartographic;
+      };
+      cartesianArray: Cartesian3[];
+    } => {
+    const viewer = cesium.current?.cesiumElement;
+    if (
+      !viewer ||
+      viewer.isDestroyed() ||
+      !property?.cameraLimiter ||
+      !property?.cameraLimiter.cameraLimitterTargetArea ||
+      !geodsic
+    )
+      return undefined;
+    const targetHalfWidth = (property.cameraLimiter.cameraLimitterTargetWidth || targetWidth) / 2;
+    const targetHalfHeight = (property.cameraLimiter.cameraLimitterTargetLength || targetHeight) / 2;
+    const topDemention = geodsic.geodesicVertical.interpolateUsingSurfaceDistance(targetHalfHeight);
+    const bottomDemention = geodsic.geodesicVertical.interpolateUsingSurfaceDistance(-targetHalfHeight);
+    const rightDemention = geodsic.geodesicHorizontal.interpolateUsingSurfaceDistance(targetHalfWidth);
+    const leftDemention = geodsic.geodesicHorizontal.interpolateUsingSurfaceDistance(-targetHalfWidth);
+
+    const rightTop = new Cartographic(rightDemention.longitude, topDemention.latitude, 0);
+    const leftTop = new Cartographic(leftDemention.longitude, topDemention.latitude, 0);
+    const rightBottom = new Cartographic(rightDemention.longitude, bottomDemention.latitude, 0);
+    const leftBottom = new Cartographic(leftDemention.longitude, bottomDemention.latitude, 0);
+
+    return {
+      cartographicDimensions: {
+        rightDemention,
+        leftDemention,
+        topDemention,
+        bottomDemention,
+      },
+      cartesianArray: [
+        Cartographic.toCartesian(rightTop),
+        Cartographic.toCartesian(leftTop),
+        Cartographic.toCartesian(leftBottom),
+        Cartographic.toCartesian(rightBottom),
+        Cartographic.toCartesian(rightTop),
+      ],
+    };
+  }, [property?.cameraLimiter, geodsic]);
+
+  // calculate maximum camera view (outer boundaries)
+  const cameraViewOuterBoundaries = useMemo((): undefined | { cartesianArray: Cartesian3[] } => {
+    const viewer = cesium.current?.cesiumElement;
+    if (
+      !viewer ||
+      !viewer.scene ||
+      viewer.isDestroyed() ||
+      !property?.cameraLimiter ||
+      !property?.cameraLimiter.cameraLimitterTargetArea ||
+      !geodsic
+    )
+      return undefined;
+
+    const camera = new CesiumCamera(viewer.scene);
+    camera.setView({
```

Comment:
Don't occur side effects in `useMemo`. Use `useEffect` instead of or present side effects as state with `useState`.

Reply:
@rot1024 , please check this chunk of code.
Thanks for your comment, I did not know that you should not have side effects inside `useMemo`
I'm using now `useEffect` for storing value as a state with `useState`

Reply:
Yes, mentioned by official reference: https://reactjs.org/docs/hooks-reference.html#usememo

> Remember that the function passed to useMemo runs during rendering. Don’t do anything there that you wouldn’t normally do while rendering. For example, side effects belong in useEffect, not useMemo.